bridges/snowbridge/pallets/system-v2/src/lib.rs

```diff
@@ -141,40 +139,36 @@ pub mod pallet {
 		/// Sends a command to the Gateway contract to instantiate a new agent contract representing
 		/// `origin`.
 		///
-		/// Fee required: Yes
-		///
-		/// - `origin`: Must be `Location` of a sibling parachain
+		/// - `location`: The location representing the agent
+		/// - `fee`: Ether to pay for the execution cost on Ethereum
 		#[pallet::call_index(1)]
 		#[pallet::weight(T::WeightInfo::create_agent())]
 		pub fn create_agent(
 			origin: OriginFor<T>,
 			location: Box<VersionedLocation>,
 			fee: u128,
 		) -> DispatchResult {
-			let _ = match origin.clone().into() {
-				Ok(RawOrigin::Root) => Ok(Here.into()),
-				_ => T::SiblingOrigin::ensure_origin(origin),
-			}?;
+			T::SiblingOrigin::ensure_origin(origin)?;
 
 			let origin_location: Location =
 				(*location).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
```

Comment:
This is confusing naming, since reviewers will assume `origin_location` corresponds with `origin` rather than `location`.

Rather just rebind `location`:
```suggestion
			let location: Location =
				(*location).try_into().map_err(|_| Error::<T>::UnsupportedLocationVersion)?;
```

Reply:
I search the code base in `polkadot-sdk` there are a lot of the usage of `origin_location` which refers to location without confusion.

So I prefer to not change.

Reply:
But the initial `location` parameter is different to `origin`. That is the source of confusion.

reviewers will think `origin_location` is derived from `origin` rather than `location`.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/6/commits/493d24312a29518e9adc8b1c3c843fbdbbd5bef9